File: Data Structures & Algorithms/products-of-array-discluding-self/submission-9.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int n = nums.Length;
        int[] res = new int[n];

        for (int i = 0; i < n; i++) {
            int prod = 1;
            for (int j = 0; j < n; j++) {
                if (i != j) {
                    prod *= nums[j];
                }
            }
            res[i] = prod;
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-9.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> dp;
        dp.push_back(nums[0]);
        int ans=1,n=nums.size();
        for(int i=1;i<n;i++){
            if(dp.back() < nums[i]){ 
                dp.push_back(nums[i]);
                ans++;
            }
            else dp[lower_bound(dp.begin(),dp.end(),nums[i]) - dp.begin()]=nums[i]; 
        }
        return ans;
    }
};
